src/NHibernate/Linq/LinqExtensionMethods.cs

```diff
@@ -8,11 +8,1582 @@
+				{"SumLong?Param", GetMethod(() => new EnumerableQuery<long?>(new long?[] { }).Sum(o => (long?)1))},
+				{"SumFloatParam", GetMethod(() => new EnumerableQuery<float>(new float[] { }).Sum(o => (float)1))},
+				{"SumFloat?Param", GetMethod(() => new EnumerableQuery<float?>(new float?[] { }).Sum(o => (float?)1))},
+				{"SumDoubleParam", GetMethod(() => new EnumerableQuery<double>(new double[] { }).Sum(o => (double)1))},
+				{"SumDouble?Param", GetMethod(() => new EnumerableQuery<double?>(new double?[] { }).Sum(o => (double?)1))},
+				{"SumDecimalParam", GetMethod(() => new EnumerableQuery<decimal>(new decimal[] { }).Sum(o => (decimal)1))},
+				{"SumDecimal?Param", GetMethod(() => new EnumerableQuery<decimal?>(new decimal?[] { }).Sum(o => (decimal?)1))},
+
+				{"AverageInt", GetMethod(() => new EnumerableQuery<int>(new int[] { }).Average())},
+				{"AverageInt?", GetMethod(() => new EnumerableQuery<int?>(new int?[] { }).Average())},
+				{"AverageLong", GetMethod(() => new EnumerableQuery<long>(new long[] { }).Average())},
+				{"AverageLong?", GetMethod(() => new EnumerableQuery<long?>(new long?[] { }).Average())},
+				{"AverageFloat", GetMethod(() => new EnumerableQuery<float>(new float[] { }).Average())},
+				{"AverageFloat?", GetMethod(() => new EnumerableQuery<float?>(new float?[] { }).Average())},
+				{"AverageDouble", GetMethod(() => new EnumerableQuery<double>(new double[] { }).Average())},
+				{"AverageDouble?", GetMethod(() => new EnumerableQuery<double?>(new double?[] { }).Average())},
+				{"AverageDecimal", GetMethod(() => new EnumerableQuery<decimal>(new decimal[] { }).Average())},
+				{"AverageDecimal?", GetMethod(() => new EnumerableQuery<decimal?>(new decimal?[] { }).Average())},
+				{"AverageIntParam", GetMethod(() => new EnumerableQuery<int>(new int[] { }).Average(o => 1))},
+				{"AverageInt?Param", GetMethod(() => new EnumerableQuery<int?>(new int?[] { }).Average(o => (int?)1))},
+				{"AverageLongParam", GetMethod(() => new EnumerableQuery<long>(new long[] { }).Average(o => (long)1))},
+				{"AverageLong?Param", GetMethod(() => new EnumerableQuery<long?>(new long?[] { }).Average(o => (long?)1))},
+				{"AverageFloatParam", GetMethod(() => new EnumerableQuery<float>(new float[] { }).Average(o => (float)1))},
+				{"AverageFloat?Param", GetMethod(() => new EnumerableQuery<float?>(new float?[] { }).Average(o => (float?)1))},
+				{"AverageDoubleParam", GetMethod(() => new EnumerableQuery<double>(new double[] { }).Average(o => (double)1))},
+				{"AverageDouble?Param", GetMethod(() => new EnumerableQuery<double?>(new double?[] { }).Average(o => (double?)1))},
+				{"AverageDecimalParam", GetMethod(() => new EnumerableQuery<decimal>(new decimal[] { }).Average(o => (decimal)1))},
+				{"AverageDecimal?Param", GetMethod(() => new EnumerableQuery<decimal?>(new decimal?[] { }).Average(o => (decimal?)1))},
+			};
+		}
+
+		private static MethodInfo GetMethod<T>(Expression<Func<T>> expression)
```

Comment:
And I have provided [here](https://github.com/nhibernate/nhibernate-core/pull/588#discussion_r119885149) a replacement example avoiding to create a dummy instance just for pointing at one of its method, removing all those constructor calls.